content/docs/pulumi-cloud/deployments/api.md

```diff
@@ -89,19 +89,25 @@ Using a custom image may result in slower execution due to time spent pulling th
 
 ### SourceContext
 
-The source context contains information about where the source code for your project is located. Currently, only git repos are supported as a source.
-
-* **repoURL** (Optional[string]): The URL of the git repository where the source code is located. Must not be specified if a [GitHub block](#github) is present.
-* **branch** (Optional[string]): The repository branch to use.
-* **repoDir** (Optional[string]): The directory where Pulumi.yaml is located, if not in the project source root.
-* **commit** (Optional[string]): The hash of the commit to deploy. If used, HEAD will be in detached mode. This is mutually exclusive with the branch setting. Either value must be specified.
-* **gitAuth** (Optional[object]): The authentication information for the git repo. If not specified, the repo is assumed to be public. Only one type is supported at a time.
-    * **sshAuth** (Optional[object]): SSHAuth is the authentication information for the git repo
-        * **sshPrivateKey** (Secret): The private key to use
-        * **password** (Optional[Secret]): The password to use
-    * **basicAuth** (Optional[object]): Basic auth information
-        * **userName** (string): The username to use for authentication
-        * **password** (Secret): The password to use for authentication
+The source context contains information about where the source code for your project is located. Currently, git repos and templates are supported as sources.
+
+* **git** (Optional[object]): Git repository information. Must not be specified if a [GitHub block](#github) is present or if a template is specified.
+    * **repoURL** (Optional[string]): The URL of the git repository where the source code is located.
+    * **branch** (Optional[string]): The repository branch to use.
+    * **repoDir** (Optional[string]): The directory where Pulumi.yaml is located, if not in the project source root.
+    * **commit** (Optional[string]): The hash of the commit to deploy. If used, HEAD will be in detached mode. This is mutually exclusive with the branch setting. Either value must be specified.
+    * **gitAuth** (Optional[object]): The authentication information for the git repo. If not specified, the repo is assumed to be public. Only one type is supported at a time.
+        * **sshAuth** (Optional[object]): SSHAuth is the authentication information for the git repo
+            * **sshPrivateKey** (Secret): The private key to use
+            * **password** (Optional[Secret]): The password to use
+        * **basicAuth** (Optional[object]): Basic auth information
+            * **userName** (string): The username to use for authentication
+            * **password** (Secret): The password to use for authentication
+
+* **template** (Optional[object]): Template information. Must not be specified if git information is present or if a [GitHub block](#github) is present.
```

Comment:
At the moment, the only thing that users can configure for the template mode is:
```
{
  "sourceType": "no-code"
}
```

This relies on the `ProjectSettings` for determining what template to use. In other words, templates are currently configured on the project level. I've cut https://github.com/pulumi/pulumi-service/issues/28437 to expose those same settings in the deployment settings as well as expose `ProjectSettings` to users (currently those are tightly tied to the NPW/NSW and its console-only APIs).